sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -638,6 +655,17 @@ private[hive] class HiveClientImpl(
             s"No partition is dropped. One partition spec '$s' does not exist in table '$table' " +
             s"database '$db'")
         }
+        // Check whether the partition we are going to drop is empty.
+        // We make a dummy one for the empty partition. See [SPARK-29786] for more details.
+        parts.foreach { partition =>
+          val partPath = partition.getPath.head
+          if (isEmptyPath(partPath)) {
```

Comment:
I think we need to check non-existing path, not empty path?

Reply:
> I think we need to check non-existing path, not empty path?

Yes, you're right. We only need to check the existence of path instead of those under the path.